cfn-signal.service and set-aws-environment.service are failing for node pools

With https://github.com/coreos/kube-aws/commit/9e618c93d723a6d02adb181635401e1f946a14dd, a node pool with a autoscaling group containing a single EC2 instance failed like:

```
REATE_FAILED AWS::CloudFormation::Stack kubeawstest1-fleet2 The following resource(s) failed to create: [AutoScaleWorker].
CREATE_FAILED AWS::AutoScaling::AutoScalingGroup AutoScaleWorker Received 0 SUCCESS signal(s) out of 1.  Unable to satisfy 100% MinSuccessfulInstancesPercent requirement
```

It seems to be caused by failures in `cfn-signal.service` and `set-aws-environment.service`:

```
CoreOS stable (1185.5.0)
Update Strategy: No Reboots
Failed Units: 2
  cfn-signal.service
  set-aws-environment.service
```

Full log from each service can be seen at:
https://gist.github.com/mumoshu/462fa1a599fea7ad3f3e82442a8c110c

--- a/config/config.go
+++ b/config/config.go
@@ -683,6 +683,12 @@ type Config struct {
 	VPCRef string
 }
 
+// CloudFormation stack name which is unique in an AWS account.
+// This is intended to be used to reference stack name from cloud-config as the target of awscli or cfn-bootstrap-tools commands e.g. `cfn-init` and `cfn-signal`
+func (c Config) StackName() string {
+	return c.ClusterName
+}
+
 func (c Cluster) valid() error {
 	if c.CreateRecordSet {
 		if c.HostedZone == "" && c.HostedZoneID == "" {

--- a/config/templates/cloud-config-worker
+++ b/config/templates/cloud-config-worker
@@ -209,7 +209,7 @@ coreos:
           {{.AWSCliImageRepo}}:{{.AWSCliTag}} -- cfn-init -v \
               --region {{.Region}} \
               --resource LaunchConfigurationWorker \
-              --stack {{.ClusterName}}
+              --stack {{.StackName}}
 {{end}}
 
 {{if .Worker.SpotFleet.Enabled}}
@@ -290,7 +290,7 @@ coreos:
           {{.AWSCliImageRepo}}:{{.AWSCliTag}} -- cfn-signal -e 0 \
               --region {{.Region}} \
               --resource AutoScaleWorker \
-              --stack {{.ClusterName}}
+              --stack {{.StackName}}
 {{end}}
 
 {{if .Experimental.AwsNodeLabels.Enabled }}

--- a/e2e/run
+++ b/e2e/run
@@ -3,7 +3,9 @@
 KUBE_AWS_CMD=${KUBE_AWS_CMD:-$GOPATH/src/github.com/coreos/kube-aws/bin/kube-aws}
 E2E_DIR=$(cd $(dirname $0); pwd)
 WORK_DIR=${E2E_DIR}/assets/${KUBE_AWS_CLUSTER_NAME}
-NODE_POOL_ASSETS_DIR=${E2E_DIR}/assets/${KUBE_AWS_CLUSTER_NAME}/node-pools/${KUBE_AWS_POOL_NAME}
+KUBE_AWS_NODE_POOL_INDEX=${KUBE_AWS_NODE_POOL_INDEX:-1}
+KUBE_AWS_NODE_POOL_NAME=${KUBE_AWS_CLUSTER_NAME}-nodepool${KUBE_AWS_NODE_POOL_INDEX}
+NODE_POOL_ASSETS_DIR=${E2E_DIR}/assets/${KUBE_AWS_CLUSTER_NAME}/node-pools/${KUBE_AWS_NODE_POOL_NAME}
 SRC_DIR=$(cd $(dirname $0); cd ..; pwd)
 KUBECONFIG=${WORK_DIR}/kubeconfig
 
@@ -71,6 +73,13 @@ configure() {
   echo 'createRecordSet: true' >> cluster.yaml
 
   # required to run kube-aws update
+  echo 'workerCount: 2' >> cluster.yaml
+  echo 'controllerCount: 2' >> cluster.yaml
+
+  if [ "${KUBE_AWS_USE_CALICO}" != "" ]; then
+    echo 'useCalico: true' >> cluster.yaml
+  fi
+
   customize_worker
 
   ${KUBE_AWS_CMD} render
@@ -82,8 +91,6 @@ configure() {
 }
 
 customize_worker() {
-  echo 'workerCount: 2' >> cluster.yaml
-  echo 'controllerCount: 2' >> cluster.yaml
   echo -e 'experimental:\n  nodeDrainer:\n    enabled: true' >> cluster.yaml
   if [ "${KUBE_AWS_WAIT_SIGNAL_ENABLED}" != "" ]; then
     echo -e '  waitSignal:\n    enabled: true' >> cluster.yaml
@@ -96,9 +103,6 @@ customize_worker() {
   fi
   if [ "${KUBE_AWS_AWS_ENV_ENABLED}" != "" ]; then
     echo -e "  awsEnvironment:\n    enabled: true\n    environment:\n      CFNSTACK: '{\"Ref\":\"AWS::StackId\"}'" >> cluster.yaml
-  fi
-  if [ "${KUBE_AWS_USE_CALICO}" != "" ]; then
-    echo 'useCalico: true' >> cluster.yaml
   fi
 }
 
@@ -224,34 +228,40 @@ ssh_worker() {
 nodepool_init() {
   cd ${WORK_DIR}
 
-  ${KUBE_AWS_CMD} node-pools init --node-pool-name ${KUBE_AWS_POOL_NAME} \
+  ${KUBE_AWS_CMD} node-pools init --node-pool-name ${KUBE_AWS_NODE_POOL_NAME} \
       --availability-zone ${KUBE_AWS_AVAILABILITY_ZONE} \
       --key-name ${KUBE_AWS_KEY_NAME} \
       --kms-key-arn ${KUBE_AWS_KMS_KEY_ARN}
+
+  cd ${NODE_POOL_ASSETS_DIR}
 
   if [ "${KUBE_AWS_SPOT_FLEET_ENABLED}" != "" ]; then
     echo Writing ${NODE_POOL_ASSETS_DIR}/cluster.yaml
     echo -e "worker:\n  spotFleet:\n    targetCapacity: 3\n" >> ${NODE_POOL_ASSETS_DIR}/cluster.yaml
   fi
+
+  echo -e "instanceCIDR: 10.0.${KUBE_AWS_NODE_POOL_INDEX}.0/24" >> ${NODE_POOL_ASSETS_DIR}/cluster.yaml
+
+  customize_worker
 }
 
 nodepool_render() {
   cd ${WORK_DIR}
 
-  ${KUBE_AWS_CMD} node-pools render stack --node-pool-name ${KUBE_AWS_POOL_NAME}
+  ${KUBE_AWS_CMD} node-pools render stack --node-pool-name ${KUBE_AWS_NODE_POOL_NAME}
 }
 
 nodepool_validate() {
   cd ${WORK_DIR}
 
-  ${KUBE_AWS_CMD} node-pools validate --node-pool-name ${KUBE_AWS_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
+  ${KUBE_AWS_CMD} node-pools validate --node-pool-name ${KUBE_AWS_NODE_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
 }
 
 nodepool_up() {
   cd ${WORK_DIR}
 
-  ${KUBE_AWS_CMD} node-pools up --node-pool-name ${KUBE_AWS_POOL_NAME} --export
-  ${KUBE_AWS_CMD} node-pools up --node-pool-name ${KUBE_AWS_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
+  ${KUBE_AWS_CMD} node-pools up --node-pool-name ${KUBE_AWS_NODE_POOL_NAME} --export
+  ${KUBE_AWS_CMD} node-pools up --node-pool-name ${KUBE_AWS_NODE_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
 }
 
 nodepool_update() {
@@ -269,13 +279,13 @@ nodepool_update() {
 
   popd
 
-  ${KUBE_AWS_CMD} node-pools update --node-pool-name ${KUBE_AWS_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
+  ${KUBE_AWS_CMD} node-pools update --node-pool-name ${KUBE_AWS_NODE_POOL_NAME} --s3-uri ${KUBE_AWS_S3_URI}
 }
 
 nodepool_destroy() {
   cd ${WORK_DIR}
 
-  ${KUBE_AWS_CMD} node-pools destroy --node-pool-name ${KUBE_AWS_POOL_NAME}
+  ${KUBE_AWS_CMD} node-pools destroy --node-pool-name ${KUBE_AWS_NODE_POOL_NAME}
 }
 
 nodepool() {

--- a/nodepool/config/config.go
+++ b/nodepool/config/config.go
@@ -214,6 +214,12 @@ func (c ProvidedConfig) valid() error {
 	return nil
 }
 
+// CloudFormation stack name which is unique in an AWS account.
+// This is intended to be used to reference stack name from cloud-config as the target of awscli or cfn-bootstrap-tools commands e.g. `cfn-init` and `cfn-signal`
+func (c ComputedConfig) StackName() string {
+	return c.NodePoolName
+}
+
 func (c ComputedConfig) VPCRef() string {
 	//This means this VPC already exists, and we can reference it directly by ID
 	if c.VPCID != "" {

